GeneratorInterface/Core/src/BaseHadronizer.cc

```diff
@@ -17,4 +31,28 @@ BaseHadronizer::BaseHadronizer( edm::ParameterSet const& ps )
 
 }
 
+void BaseHadronizer::randomizeIndex(edm::LuminosityBlock const& lumi, CLHEP::HepRandomEngine* rengine) {
+  if (randomInitWeights_.size()>0) {
+    //randomly select from a list of provided configuration sets (for parameter scans)
+    
+    //seeds std 32-bit mersene twister with 50 random integers plus run and lumi section numbers
```

Comment:
@Dr15Jones this is a further updated implementation on how to generate random numbers deterministically in beginLuminosityBlock.  In this case an independent mersene twister generator is initialized using random numbers from the CMSSW random number engine combined with the run and lumi section numbers.